src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -28,14 +28,27 @@ import {
 	DEV_ENVIRONMENT_PHP_VERSIONS,
 } from '../constants/dev-environment';
 import { getVersionList, readEnvironmentData } from './dev-environment-core';
-import type { AppInfo, ComponentConfig, InstanceOptions, EnvironmentNameOptions, InstanceData } from './types';
+import type {
+	AppInfo,
+	ComponentConfig,
+	InstanceOptions,
+	EnvironmentNameOptions,
+	InstanceData,
+	WordPressConfig,
+} from './types';
 import { validateDockerInstalled, validateDockerAccess } from './dev-environment-lando';
 import UserError from '../user-error';
+import typeof Command from 'lib/cli/command';
 
 const debug = debugLib( '@automattic/vip:bin:dev-environment' );
 
 const DEFAULT_SLUG = 'vip-local';
 
+// Forward declaratrion to avoid no-use-before-define
+declare function promptForComponent( component: 'wordpress', allowLocal: false, defaultObject: ComponentConfig | null ): Promise<WordPressConfig>;
```

Comment:
Maybe let's have a different function for WordPress vs other components (where you get to choose the mode) altogether.

Apart from cleaner type definitions, the functions would get simpler too.

What do you think?

Reply:
Yes, I agree. But this requires more changes to the code, so I left that for another PR to make it easier to review this one.